msteams-platform/bots/how-to/conversations/conversation-messages.md

```diff
@@ -443,16 +443,31 @@ For more information on cards and cards in bots, see [cards documentation](~/tas
-| 403 | **Code**: `BotDisabledByAdmin` <br/> **Message**: The tenant admin disabled this bot. | Tenant blocked the bot. |
-| 401 | **Code**: `BotNotRegistered` <br/> **Message**: No registration found for this bot. | The registration for this bot wasn't found. |
-| 412 | **Code**: `PreconditionFailed` <br/> **Message**: Precondition failed, please try again. | A precondition failed on one of our dependencies due to multiple concurrent operations on the same conversation. |
-| 404 | **Code**: `ConversationNotFound` <br/> **Message**: Conversation not found. | The conversation wasn't found. |
-| 413 | **Code**: `MessageSizeTooBig` <br/> **Message**: Message size too large. | The size on the incoming request was too large. |
-| 429 | **Code**: `Throttled` <br/> **Message**: Too many requests. Also returns when to retry after. | Too many requests were sent by the bot. For more information, see [rate limit](~/bots/how-to/rate-limit.md). |
+| Status code | Error code and message values | Description | Should retry request | Developer Action |
+|----------------|-----------------|-----------------|----------------|----------------|
+| 403 | **Code**: `ConversationBlockedByUser` <br/> **Message**: User blocked the conversation with the bot. | User blocked the bot in 1:1 chat or a channel through moderation settings. | No | Remove conversation from cache. Stop attempting to post conversation until interaction with bot is explicitly initiated by a user in the conversation, indicating bot is no longer blocked. |
+| 403 | **Code**: `BotNotInConversationRoster` <br/> **Message**: The bot isn't part of the conversation roster. | The bot isn't part of the conversation. App needs to be reinstalled in conversation. | No | Wait for installationUpdate event [Conversation events Teams Microsoft Docs](/microsoftteams/platform/bots/how-to/conversations/subscribe-to-conversation-events?tabs=dotnet) indicating that the bot has been readded before attempting to send conversation requests. |
+| 403 | **Code**: `BotDisabledByAdmin` <br/> **Message**: The tenant admin disabled this bot. | User to bot interactions are blocked by tenant admins. Tenant admins need to allow the app for the user. See [app policies](/microsoftteams/app-policies). | No | Stop attempting to post conversation until interaction with bot is explicitly initiated by a user in the conversation, indicating bot is no longer blocked. |
+| 403 | **Code**: `NotEnoughPermissions` <br/> **Message**: *scenario specific. | Bot lacks necessary permissions to perform the requested action. | No | Determine action from error message. |
+| 401 | **Code**: `BotNotRegistered` <br/> **Message**: No registration found for this bot. | The registration for this bot wasn't found. | No | Verify the bot ID and password used. Ensure bot ID is registered with Microsoft through the Bot Channel Registration in Azure (must manually enable "Teams" bot channel), Teams Developer Portal, or Teams App Studio (under deprecation). |
```

Comment:
> Verify the bot ID and password used.

How would an incorrect password lead to this error-- wouldn't the app have failed to get an access token?